src/loose/statement.js

```diff
@@ -329,7 +330,8 @@ lp.parseFunction = function(node, isStatement, isAsync) {
   else if (isStatement === true) node.id = this.dummyIdent()
   this.inAsync = node.async
   node.params = this.parseFunctionParams()
-  node.body = this.parseBlock()
+  this.parseFunctionBody(node)
```

Comment:
This seems to allow brace-less function bodies for regular functions — I don't think that logic needs to be in `parseFunctionBody`, since it only applies to arrow functions.

Reply:
`parseMethod` has the behavior, can parse the body as an expression, but can I remove it?

https://github.com/ternjs/acorn/pull/593/files#diff-d91a17a6a9019eba5785445f28c72421L518

Reply:
Yeah, that doesn't seem to make a lot of sense